src/isomorphism.rs

```diff
@@ -28,9 +29,151 @@ use petgraph::visit::{
     EdgeRef, GetAdjacencyMatrix, IntoEdgeReferences, NodeIndexable,
 };
 use petgraph::EdgeType;
-use petgraph::{Directed, Incoming};
+use petgraph::{Directed, Incoming, Outgoing};
 
-impl<'a, Ty> NodesRemoved for &'a StableGraph<PyObject, PyObject, Ty>
+type Graph<Ty> = StableGraph<PyObject, PyObject, Ty>;
```

Comment:
I used `Graph` since we assume contiguous node ids in many places. but i get that it might be confusing. if you have a good suggestion for the name,  i will use it

Reply:
What do you think about `GenericPyGraph` or something like that? Basically the distinction here is that the weight types are `PyObjects`.

Reply:
Oh or maybe `StablePyGraph` (to indicate it's a stable graph with pyobject weights)?

Reply:
yes, `StablePyGraph` is fine.